### PR TITLE
Improve CI failure detection

### PR DESCRIPTION
**Pull request by joestringer**
Fix a few issues noted during an attempt to debug a CI issue:
* k8s policy waiting was ignoring failures in the command
* L7 proxy redirect test wasn't waiting for endpoints to be ready

Related: #3470

### CODE REVIEW

**Comment by joestringer** [comment]
test-me-please

File: test/helpers/kubectl.go [hunk 1]
```diff
@@ -899,12 +902,12 @@ func (kub *Kubectl) DumpCiliumCommandOutput(namespace string) {
...
-		"kubectl get ds -o wide --all-namespaces":       "ds.txt",
-		"kubectl get cnp --all-namespaces":              "cnp.txt",
-		"kubectl describe pods --all-namespaces":        "pods_status.txt",
-		"kubectl -n kube-system logs -l k8s-app=cilium": "cilium_logs.txt",
+		"kubectl get pods -o wide --all-namespaces":                  "pods.txt",
+		"kubectl get services -o wide --all-namespaces":              "svc.txt",
+		"kubectl get ds -o wide --all-namespaces":                    "ds.txt",
+		"kubectl get cnp --all-namespaces":                           "cnp.txt",
+		"kubectl describe pods --all-namespaces":                     "pods_status.txt",
+		"kubectl -n kube-system logs -l k8s-app=cilium --timestamps": "cilium_logs.txt",
```
Comment (ianvernon) [inline test/helpers/kubectl.go]: 6961a5daedeb49ce738e8d42006a7e68a0818eb2 removed the usage of timestamps - might want to remove this for now and then file a GH issue to explore whether we can use `--timestamps` in the future.

**Review by ianvernon — DISMISSED** [review]
Please remove the re-adding of `--timestamps` and file a separate GH issue for it since we have been having issues with it recently.

File: test/runtime/ct.go [hunk 2]
```diff
@@ -725,8 +725,11 @@ var _ = Describe("DisabledRuntimeValidatedConntrackTable", func() {
 	})
 
 	It("Testing L7 proxy redirection after a L3 connection is made with the exact 5 tuple", func() {
-		vm.SetPolicyEnforcement(helpers.PolicyEnforcementDefault)
-		vm.WaitEndpointsReady()
+		res := vm.SetPolicyEnforcement(helpers.PolicyEnforcementDefault)
+		res.ExpectSuccess("Setting policyEnforcement to default")
+		res := vm.WaitEndpointsReady()
+		Expect(areEndpointsReady).Should(BeTrue(), "Endpoints are not ready after timeout")
```
Comment (ianvernon) [inline test/runtime/ct.go]: Copy-paste error ;)

File: test/helpers/kubectl.go (hunk 1 above)
Comment (joestringer) [reply test/helpers/kubectl.go]: @ianvernon I think you linked the wrong commit? We're using timestamps elsewhere, there was temporarily a problem when `--timestamps` was specified before the command (ie, `logs`), but after we shuffled that around I don't think we've seen problems, have we?

File: test/runtime/ct.go (hunk 2 above)
Comment (joestringer) [reply test/runtime/ct.go]: beh. Thanks. -_-

**Comment by joestringer** [comment]
test-me-please

**Comment by joestringer** [comment]
test-me-please

File: test/helpers/kubectl.go [hunk 3]
```diff
@@ -736,7 +736,10 @@ func (kub *Kubectl) CiliumPolicyAction(namespace, filepath string, action Resour
 			// Wait until all the pods are synced
 			for pod, rev := range waitingRev {
 				kub.logger.Infof("CiliumPolicyAction: Wait for endpoints to sync on pod '%s'", pod)
-				kub.ExecPodCmd(namespace, pod, fmt.Sprintf("cilium policy wait %d", rev))
+				res := kub.ExecPodCmd(namespace, pod, fmt.Sprintf("cilium policy wait %d", rev))
```
Comment (eloycoto) [inline test/helpers/kubectl.go:739]: Good one!

**Review by eloycoto — APPROVED** [review]

**Comment by joestringer** [comment]
test-me-please

File: test/helpers/kubectl.go (hunk 1 above)
Comment (ianvernon) [reply test/helpers/kubectl.go]: Sorry about that, did link to the wrong commit. You are right, it appears that it was the ordering that was the issue, not the `--timestamps` parameter itself.

**Review by ianvernon — APPROVED** [review]

**Pull request merged**